src/services/graphToState.ts

```diff
@@ -0,0 +1,26 @@
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { lifecycle, shallowEqual } from 'recompose';
+import { graphToState, GraphToStateAction } from '../reducers/rootReducer';
+
+/**
+ * A simple HoC that can be bound around a component that's wrapped with `react-apollo`'s `graphql`.
+ * This will a) allow us to keep using functional components and b) cut down on boilerplate code like connecting the reducer, and dispatching data to the state once it's ready.
+ * It also sets a global uniform way to move data from the graph to the state.
+ * @param {string} action Action that will be dispatched whenever the data prop changes.
+ * @param {string} dataProp Prop that will be dispatched to reducer. By default it's data, but this option can be used in case it got remapped to something else (e.g. 'metadata')
+ */
+export default <Props>(action: GraphToStateAction['type'], dataProp: keyof Props) =>
+    compose(
+        connect(null, dispatch => ({ graphToState: (payload: any) => dispatch(graphToState(action, payload)) })),
+        lifecycle({
+            componentWillReceiveProps(nextProps: Props) {
```

Comment:
This looks like it could cause problems with both view- and facetConfig. 
They really only want to set the state on mount. If Apollo tries to update anything (could happen because connected selects could create a new call to same collection) before saving, all progress is lost.

Reply:
Generally, apollo will only send new props if the data actually changed, e.g. the connected select sends a call, and the data from the server differs from the local data.
I'm fine with adding a additional `update` true/false option to the HoC, but shouldn't we start thinking about what to do when the data on the server has changed? 
We'd blindly override the server side data, despite knowing it was changed...